### PR TITLE
[SIM_FLUTTER-122] [REFACTOR] Convert TransactionEnum to Enhanced Enum

### PR DESCRIPTION
**Pull request by IanUrriza10**
### Issue Links
- [backlog](https://framgiaph.backlog.com/view/SIM_FLUTTER-122)
- [slack](https://sun-philippine.slack.com/archives/C059L7XAXFB/p1689739155201799)

### Definition of done
- [x] Working Transaction Related functions
 
### Screenshots

image

https://github.com/framgia/sph-flutter/assets/114897466/f4371248-9557-4156-a6c1-4f5a6e0ec3dc

### CODE REVIEW

File: frontend/lib/src/enums/transaction_enum.dart [hunk 1]
```diff
@@ -3,76 +3,26 @@
...
-      return TransactionCategories.BILLS;
-    case 'Savings':
-      return TransactionCategories.SAVINGS;
-    case 'Miscellaneous':
-      return TransactionCategories.MISC;
-    default:
-      return TransactionCategories.MISC;
-  }
+  static TransactionCategories fromValue(String jsonValue) =>
+      TransactionCategories.values.singleWhere((i) => jsonValue == i.value);
```
Comment (jstephend-sun) [inline frontend/lib/src/enums/transaction_enum.dart]:
```suggestion
  static TransactionCategories fromJson(String jsonValue) =>
      TransactionCategories.values.singleWhere((i) => jsonValue == i.name);
      
       static TransactionCategories fromValue(String value) =>
      TransactionCategories.values.singleWhere((i) => value == i.value);
```

then add some comments on the static constructor

File: frontend/lib/src/enums/transaction_enum.dart [hunk 2]
```diff
@@ -3,76 +3,26 @@
 enum TransactionTypes { CREDIT, DEPT, TRANSFER }
```
Comment (jstephend-sun) [inline frontend/lib/src/enums/transaction_enum.dart]: apply refactor to this as well

File: frontend/lib/src/enums/transaction_enum.dart [hunk 3]
```diff
@@ -3,76 +3,26 @@
...
-    case 'Housing/Billings':
-      return TransactionCategories.BILLS;
-    case 'Savings':
-      return TransactionCategories.SAVINGS;
-    case 'Miscellaneous':
-      return TransactionCategories.MISC;
-    default:
-      return TransactionCategories.MISC;
-  }
+  static TransactionCategories fromValue(String jsonValue) =>
```
Comment (jstephend-sun) [inline frontend/lib/src/enums/transaction_enum.dart]: apply fromJson to the model

for type and category

image

File: frontend/lib/src/enums/transaction_enum.dart [hunk 4]
```diff
@@ -3,76 +3,26 @@
...
-      return '';
-  }
-}).toList();
+  static List<String> creditCategories = [
+    TransactionCategories.FOOD.value,
+    TransactionCategories.TRANSPORTATION.value,
+    TransactionCategories.BILLS.value,
+    TransactionCategories.SAVINGS.value,
+    TransactionCategories.MISC.value,
+  ].toList();
```
Comment (jstephend-sun) [inline frontend/lib/src/enums/transaction_enum.dart]: unnecessary toList

**Review by jstephend-sun — CHANGES_REQUESTED** [review]
Requested Changes #77

review time: 10mins
comments: 4
iteration: 1st review

**Review by jstephend-sun — APPROVED** [review]
Approved #77

review time: 10mins
comments: 4
iteration: 1st review

**Review by Dbulawan13 — APPROVED** [review]
Approve PR #77 

Working:
- Still able to withdraw, deposit and transfer
- Transaction history still display dynamic data

https://github.com/framgia/sph-flutter/assets/86758474/b71f1d34-8cc7-4869-af25-e1cb8382cf44

Testing hours: 0.23
Comment: 0
Iteration: 0

**Pull request merged**